File: test/core/lib/sink.cc
#include <omnetpp.h>

using namespace omnetpp;

namespace testlib {

class Sink : public cSimpleModule
{
  protected:
    int numPackets;

  protected:
    virtual void initialize() override;
    virtual void handleMessage(cMessage *msg) override;
    virtual void finish() override;
};

Define_Module(Sink);

void Sink::initialize()
{
    EV << "Initializing Sink " << getFullPath() << "\n";
    numPackets = 0;
}

void Sink::handleMessage(cMessage *msg)
{
    EV << "Sink received msg " << msg->getName() << "\n";
    numPackets++;
    delete msg;
}

void Sink::finish()
{
    EV << "Sink finishing, received msgs: " << numPackets << "\n";
}

}

File: contrib/jsimplemodule/jsamples/src/jsamples/Sink.java
package jsamples;

import org.omnetpp.simkernel.*;

/**
 * Just discards messages.
 */
public class Sink extends JSimpleModule {

    public void handleMessage(cMessage msg) {
        ev.println(msg.getName()+" arrived at sink");
        msg.delete();
    }

    protected void finish() {
        ev.println("finish of "+getFullPath());
    }
}
